### PR TITLE
agents: track indexing tasks

### PR DESCRIPTION
**Pull request by mtanzim**
This PR introduces a new feature for agent task management and makes changes to the backend and frontend of the application. 

## Summary of Changes
- **Backend**: 
   - Added a new table, `agent_tasks`, to store agent tasks. 
   - Introduced new functions, `create_agent_task` and `get_agent_tasks_by_agent_id`, for creating and retrieving agent tasks. 
   - Modified the `handle_google_drive_sync` function to include an additional `agent_id` parameter. 
   - Updated the functions `create`, `delete`, `edit`, `move`, `rename`, `restore`, and `permission_change` to include an `agent_id` parameter and added error handling. 
   - Added a new `MinimalBase` class to the `src/backend/database_models/base.py` module. 
   - Created a new `AgentTaskResponse` model to represent agent task responses. 
   - Modified the `Agent` model to include a new `agent_tasks` attribute. 
- **Frontend**: 
   - Added new components and hooks for displaying agent tasks, including a new page, `src/interfaces/assistants_web/src/app/tasks/[agentId]/page.tsx, for viewing tasks and a right-panel link, `src/interfaces/assistants_web/src/components/Agents/RightPanel.tsx, for accessing task dashboards. 
   - Updated the `AgentTaskResponse` model to include additional properties. 
   - Introduced a new `useAgentTasks` hook for fetching agent tasks.

### CODE REVIEW

**Review by scott-cohere — APPROVED** [review]

File: src/backend/alembic/versions/2024_08_21_e284b705fb23_.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+"""empty message
```
Comment (tianjing-li) [inline src/backend/alembic/versions/2024_08_21_e284b705fb23_.py:1]: What happened with this migration? Did we have to merge heads for `('0091b3b3b4dc', 'c301506b3676')` ?

Comment (tianjing-li) [reply src/backend/alembic/versions/2024_08_21_e284b705fb23_.py:1]: Can you try deleting your migrations, pulling from main, `make reset-db` and remake your migrations

File: src/backend/services/agent.py [hunk 2]
```diff
@@ -42,3 +46,25 @@ def raise_db_error(e: Exception, type: str, name: str):
...
+def parse_task(t: SyncCeleryTaskMeta) -> AgentTaskResponse:
+    result = None
+    exception_snippet = None
+    if t.status == "SUCCESS":
+        result = pickle.loads(t.result)
+    if t.status == "FAILURE":
+        trace_lines = t.traceback.split("\n")
+        if len(trace_lines) >= 2:
+            # first 200 characters of the exception
+            exception_snippet = trace_lines[-2][:200] + "...check logs for details"
```
Comment (tianjing-li) [inline src/backend/services/agent.py:60]: Suggestion: move the `200` to some `TASK_EXCEPTION_MAX_LENGTH` (or any name that fits) constant at the top of this file

**Review by tianjing-li — CHANGES_REQUESTED** [review]
Backend changes look good to me, mostly just want the alembic migration to be regenerated.

Note that the cost of tracking tasks in SQL is you might run into scalability issues later on with the volume of writes, but if that's understood then its good to go.

File: src/backend/alembic/versions/2024_08_21_e284b705fb23_.py (hunk 1 above)
Comment (scott-cohere) [reply src/backend/alembic/versions/2024_08_21_e284b705fb23_.py:1]: this was the issue i believe
https://stackoverflow.com/a/51456404

Comment (scott-cohere) [reply src/backend/alembic/versions/2024_08_21_e284b705fb23_.py:1]: let me try doing what u said and see if it works

Comment (scott-cohere) [reply src/backend/alembic/versions/2024_08_21_e284b705fb23_.py]: that fixed it, ty

File: pyproject.toml [hunk 3]
```diff
@@ -116,7 +116,10 @@ requires = ["poetry-core"]
 build-backend = "poetry.core.masonry.api"
 
 [tool.pyright]
-include = ["src/backend/services/metrics.py"]
+include = [
+  "src/backend/services/metrics.py", 
+  "src/backend/tools/google_drive/sync/actions/",
```
Comment (giannis2two) [inline pyproject.toml:121]: What does this help with, just curious?

File: src/backend/database_models/agent_task.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    queue: Mapped[str] = mapped_column(String(155))
+
+
+class AgentTask(Base):
+    __tablename__ = "agent_tasks"
+
+    agent_id: Mapped[str] = mapped_column(
+        ForeignKey("agents.id", ondelete="CASCADE"), nullable=False
+    )
+    task_id: Mapped[str] = mapped_column(Text, nullable=False)
```
Comment (giannis2two) [inline src/backend/database_models/agent_task.py:38]: Should we make this a FK?

File: src/backend/tools/google_drive/sync/actions/utils.py [hunk 5]
```diff
@@ -17,6 +23,36 @@
 )
 from backend.tools.utils import download
 
+logger = LoggerFactory().get_logger()
+
+
+# TODO: fix type errors here
+def init_compass():
```
Comment (giannis2two) [inline src/backend/tools/google_drive/sync/actions/utils.py:30]: Let's move this outside of actions utils. This is a Compass general util?

File: src/backend/tools/google_drive/sync/actions/utils.py [hunk 6]
```diff
@@ -17,6 +23,36 @@
...
+    return Compass(
+        compass_api_url=Settings().compass.api_url,  # type: ignore
+        compass_parser_url=Settings().compass.parser_url,  # type: ignore
+        compass_username=Settings().compass.username,  # type: ignore
+        compass_password=Settings().compass.password,  # type: ignore
+    )
+
+
+def persist_agent_task(method):
```
Comment (giannis2two) [inline src/backend/tools/google_drive/sync/actions/utils.py:39]: Same with this, agent task specific util?

File: src/backend/tools/google_drive/sync/activity.py [hunk 7]
```diff
@@ -24,6 +23,7 @@
 )
 from backend.tools.google_drive.tool import GoogleDrive
 
+RESTORE_ACTION_NAME="restore"
```
Comment (giannis2two) [inline src/backend/tools/google_drive/sync/activity.py:26]: can move inside the match block for restore

**Review by giannis2two — APPROVED** [review]

**Comment by giannis2two** [comment]
Do we have a video of the UI changes and how they look? Just curious

File: pyproject.toml (hunk 3 above)
Comment (scott-cohere) [reply pyproject.toml:121]: i believe tanzim introduced this pyright static type checking. so everything included here would be checked

File: src/backend/tools/google_drive/sync/activity.py (hunk 7 above)
Comment (scott-cohere) [reply src/backend/tools/google_drive/sync/activity.py:26]: feel like it makes sense to declare the const here. not sure what u mean by moving into the block, like just passing the string directly in the param?

will leave for now to merge pr. lmk if u want me to follow up

File: src/backend/database_models/agent_task.py (hunk 4 above)
Comment (scott-cohere) [reply src/backend/database_models/agent_task.py]: ok few findings.

its seems to be a bit tricking referring to files in the celery table during a migration. it seems like this is largely due to the fact that the celery table is not created as part of our migration system. 

i tried looking into ways to work around but i dont think its worth it for now. this is working as is and we can revisit this later (or we potentially move all this to redis regardless).

**Review by tianjing-li — APPROVED** [review]

File: src/interfaces/assistants_web/src/app/tasks/[agentId]/page.tsx [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+};
+
+const Page = ({ params }: Props) => {
+  const agentId = params.agentId;
+  const {
+    data: agentTasks,
+    isLoading: isTaskLoading,
+    error: isTaskError,
+  } = useAgentTasks({ agentId });
+  const { data: agent, isLoading, error } = useAgent({ agentId });
```
Comment (knajjars) [inline src/interfaces/assistants_web/src/app/tasks/[agentId]/page.tsx:19]: this component can be changed to RSC and either hydrate these queries (see other pages for example). Or directly call the agentTasks and agent endpoint and pass the values to the children component

File: src/interfaces/assistants_web/src/app/tasks/[agentId]/page.tsx [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+        <h1 className="mb-4 mt-4 bg-slate-800 p-16 text-2xl">Tasks report for {agent?.name} </h1>
+        <h1 className="mb-4 mt-4 bg-slate-800 p-16 text-xl">No tasks yet </h1>
+      </div>
+    );
+  }
+
+  return (
+    <div className="m-12">
+      <h1 className="mb-4 mt-4 bg-slate-800 p-16 text-2xl">Tasks report for {agent?.name} </h1>
+      <table className="table-fixed border-separate border-spacing-2  border border-r-4 border-slate-500 bg-slate-800 p-8">
```
Comment (knajjars) [inline src/interfaces/assistants_web/src/app/tasks/[agentId]/page.tsx:41]: we have a Table component that we can use I believe

**Review by knajjars — APPROVED** [review]

**Pull request merged**